src/server/serveractiveobjectmap.cpp

```diff
@@ -151,28 +159,37 @@ std::unordered_set<u16> ServerActiveObjectMap::getObjectsNearBox(const aabb3s16
 	return result;
 }
 
+void ServerActiveObjectMap::addObjectRef(u16 id, v3s16 pos)
+{
+	refmap.emplace(pos, id);
+}
+
+void ServerActiveObjectMap::removeObjectRef(u16 id, v3s16 pos)
+{
+	auto bounds = refmap.equal_range(pos);
+	for (auto iter = bounds.first; iter != bounds.second;)
+		if (iter->second == id)
+			refmap.erase(iter++);
```

Comment:
I'm not sure whether C++ guarantees this to work in this form, usually this is done using:
`iter = refmap.erase(iter);`

Reply:
We're using this method in multiple locations of our code already (or still?). Using the return value is however a better/cleaner way.

Reply:
it works well on post increment. Pre increment not

Reply:
can you use std::remove_if instead ? we are using in C++11 and this is more safe :)

Reply:
I know that it works, my concern was whether C++ guarantees this.
Anyway, it's fine to keep it like this.